universe/Condition.cpp

```diff
@@ -4956,14 +4958,27 @@ void DesignHasPartClass::Eval(const ScriptingContext& parent_context,
     }
 }
 
-bool DesignHasPartClass::RootCandidateInvariant() const
-{ return (m_low->RootCandidateInvariant() && m_high->RootCandidateInvariant()); }
+bool DesignHasPartClass::RootCandidateInvariant() const { 
+    bool ret = true;
+
+    if (m_low) {
```

Comment:
rework this to avoid repeat calls to ->RootCandidateInvariant(). eg.

    return (!m_low || m_low->RootCandidateInvariant()) && (!m_high || m_high->RootCandidateInvariant());